src/test/scala/org/apache/spark/sql/delta/OptimisticTransactionSuite.scala

```diff
@@ -635,6 +635,47 @@ class OptimisticTransactionSuite extends QueryTest with SharedSparkSession {
     }
   }
 
+  test("allow read+append against data rearrange") {
+    withLog(addA_P1 :: addB_P1 :: Nil) { log =>
+      val tx1 = log.startTransaction()
+      tx1.filterFiles()
+
+      val tx2 = log.startTransaction()
+      tx2.filterFiles()
+      // tx2 rearranges files
+      tx2.commit(
+        setDataChangeFalse(addA_P1.remove :: addB_P1.remove :: addC_P1 :: Nil),
+        ManualUpdate)
+
+      tx1.commit(
+        addE_P3 :: Nil,
+        ManualUpdate)
+
+      checkAnswer(
+        log.update().allFiles.select("path"),
+        Row(C_P1) :: Row(E_P3) ::  Nil)
+    }
+  }
+
+  test("bloc delete against data rearrange") {
```

Comment:
block

Reply:
Hi @mukulmurthy, thank you for your response. No problem :)